### PR TITLE
Add option to spirv-opt to skip the validator.

### PR DESCRIPTION
**Pull request by s-perron**
The option --skip-validation is added.  When used it will skip the
validation step.

Also rearranged the help text to fix two options that are out of order.

Implements issue #1068.

### CODE REVIEW

File: tools/opt/opt.cpp [hunk 1]
```diff
@@ -463,22 +470,24 @@ int main(int argc, const char** argv) {
...
-  if (error) {
-    spvDiagnosticPrint(diagnostic);
+  if (!skip_validator) {
+    // Let's do validation first.
+    spv_context context = spvContextCreate(target_env);
+    spv_diagnostic diagnostic = nullptr;
+    spv_const_binary_t binary_struct = {binary.data(), binary.size()};
+    spv_result_t error =
+        spvValidateWithOptions(context, options, &binary_struct, &diagnostic);
+    if (error) {
```
Comment (atgoo) [inline tools/opt/opt.cpp]: Consider
```
  if (error) {
      spvDiagnosticPrint(diagnostic);
  }
  spvDiagnosticDestroy(diagnostic);
  spvValidatorOptionsDestroy(options);
  spvContextDestroy(context) 
  return error;
```

Comment (s-perron) [reply tools/opt/opt.cpp]: The way the diff looks is confusing, but we want to `return error` only if there was an error.  We would have to do

```
if (error) {
  spvDiagnosticPrint(diagnostic);
}
  spvDiagnosticDestroy(diagnostic);
  spvValidatorOptionsDestroy(options);
  spvContextDestroy(context) 
if (error) {
  return error;
}
```
I think the extra `if` is not very nice, and it only saves 1 line of code.

File: tools/opt/opt.cpp [hunk 2]
```diff
@@ -208,6 +206,11 @@ Options (in lexicographical order):
                blank spaces, and in each pair, spec id and default value must
                be separated with colon ':' without any blank spaces in between.
                e.g.: --set-spec-const-default-value "1:100 2:400"
+  --skip-validation
+               Will not validate the SPIR-V before optimizing.  If the SPIR-V
+               is invalid, the optimizer may fail or generate incorrect output.
```
Comment (dnovillo) [inline tools/opt/opt.cpp]: Add a warning that states that this should be used with care and only in special circumstances.

File: tools/opt/opt.cpp (hunk 1 above)
Comment (atgoo) [reply tools/opt/opt.cpp]: Sorry, I thought it was the end of the function.

**Pull request merged**